### PR TITLE
Adds tested null safe code excerpts

### PR DESCRIPTION
**Pull request by RedBrogdon**
* Adds a new `null_safety_examples` folder to the repo for null safe code examples.
  - Also adds two examples to the directory.
* Adds a flag to the testing scripts (code_check.sh on down) to allow for testing of these new examples.
  - Also removes a "legacy" flag that was no longer in use.
* Updates travis configuration to call scripts with the flag on the beta and dev channels.
* Adds a code excerpt to null-safety.md as a way to make sure excerpting of null safe code is working.

This PR does not touch the automatic extraction and testing of snippets as described in `/tool/extract.dart`. The comments in that file indicated that it's a deprecated pattern and new/updated snippets are meant to use the newer code excerpt packages, so I focused my effort on them.

### CODE REVIEW

**Comment by RedBrogdon** [comment]
@sfshaza2, I'm filing this now so I don't forget about it. Please do not review it until you're officially back from holiday. 🎄

File: null_safety_examples/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+1. `flutter create --no-overwrite $PROJECT`
+1. `cd $PROJECT`
+1. `flutter analyze`
+1. `flutter test`
+1. `flutter run`
+
+To learn more about setting up Flutter and running apps, see
+[flutter.dev/get-started][].
+
+[flutter.dev/get-started]: https://flutter.dev/docs/get-started
```
Comment (sfshaza2) [inline null_safety_examples/README.md:16]: A relative link should be sufficient and also works better if you've forked the repo:   `/docs/get-started`.

File: null_safety_examples/basics/lib/main.dart [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// Copyright 2018 The Flutter team. All rights reserved.
```
Comment (sfshaza2) [inline null_safety_examples/basics/lib/main.dart:1]: If this is a new file, shouldn't the copyright be 2021?

File: null_safety_examples/layout/base/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+A new Flutter project.
+
+## Getting Started
+
+This project is a starting point for a Flutter application.
+
+A few resources to get you started if this is your first Flutter project:
+
+- [Lab: Write your first Flutter app](https://flutter.dev/docs/get-started/codelab)
```
Comment (sfshaza2) [inline null_safety_examples/layout/base/README.md:11]: Use relative links:  /docs/get-started/codelab and /docs/cookbook.

File: null_safety_examples/layout/base/README.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+- [Cookbook: Useful Flutter samples](https://flutter.dev/docs/cookbook)
+
+For help getting started with Flutter, view our
+[online documentation](https://flutter.dev/docs), which offers tutorials,
```
Comment (sfshaza2) [inline null_safety_examples/layout/base/README.md:15]: Relative:  /docs

File: null_safety_examples/layout/base/lib/main.dart [hunk 5]
```diff
@@ -0,0 +1,29 @@
+// Copyright 2018 The Flutter team. All rights reserved.
```
Comment (sfshaza2) [inline null_safety_examples/layout/base/lib/main.dart:1]: Ditto on the copyright date/

File: null_safety_examples/layout/base/pubspec.yaml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+name: layout
+description: >
+  Sample app from "Building Layouts", https://flutter.io/docs/development/ui/layout.
```
Comment (sfshaza2) [inline null_safety_examples/layout/base/pubspec.yaml:3]: flutter.io?  That's an old URL.. Also, relative link is best.

File: null_safety_examples/layout/base/test/widget_test.dart [hunk 7]
```diff
@@ -0,0 +1,11 @@
+// Basic Flutter widget test. Learn more at https://flutter.io/docs/testing.
```
Comment (sfshaza2) [inline null_safety_examples/layout/base/test/widget_test.dart:1]: Ditto

File: pubspec.yaml [hunk 8]
```diff
@@ -6,11 +6,9 @@ environment:
 dev_dependencies:
   build_runner: ^1.1.2
   code_excerpt_updater:
-    git: https://github.com/chalin/code_excerpt_updater.git
-    # path: ../code_excerpt_updater
+    git: https://github.com/RedBrogdon/code_excerpt_updater.git
```
Comment (sfshaza2) [inline pubspec.yaml:9]: Thank you for this, though can't this tool live in a flutter-owned repo?

**Review by sfshaza2 — APPROVED** [review]
Exciting! I did have some comments, but I'll have to trust you on the code. :)

File: pubspec.yaml (hunk 8 above)
Comment (RedBrogdon) [reply pubspec.yaml:9]: Yes. We just nee to pick one, really. Would the site-shared repo be a good place?

CC: @kwalrath

File: null_safety_examples/README.md (hunk 1 above)
Comment (RedBrogdon) [reply null_safety_examples/README.md:16]: Done.

File: null_safety_examples/basics/lib/main.dart (hunk 2 above)
Comment (RedBrogdon) [reply null_safety_examples/basics/lib/main.dart:1]: Done.

File: null_safety_examples/layout/base/README.md (hunk 3 above)
Comment (RedBrogdon) [reply null_safety_examples/layout/base/README.md:11]: Done.

File: null_safety_examples/layout/base/README.md (hunk 4 above)
Comment (RedBrogdon) [reply null_safety_examples/layout/base/README.md:15]: Done.

File: null_safety_examples/layout/base/lib/main.dart (hunk 5 above)
Comment (RedBrogdon) [reply null_safety_examples/layout/base/lib/main.dart:1]: Done.

File: null_safety_examples/layout/base/pubspec.yaml (hunk 6 above)
Comment (RedBrogdon) [reply null_safety_examples/layout/base/pubspec.yaml:3]: Done.

File: null_safety_examples/layout/base/test/widget_test.dart (hunk 7 above)
Comment (RedBrogdon) [reply null_safety_examples/layout/base/test/widget_test.dart:1]: Done.

File: pubspec.yaml (hunk 8 above)
Comment (kwalrath) [reply pubspec.yaml:9]: site-shared sgtm

**Pull request merged**